### PR TITLE
feat(module:avatar): add avatar component

### PR DESCRIPTION
**Pull request by cipchk**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/avatar/nz-avatar.component.ts [hunk 1]
```diff
@@ -60,12 +62,12 @@ export class NzAvatarComponent implements OnChanges {
     this._classList.forEach(_className => {
       this._renderer.addClass(this._el, _className);
```
Comment (vthinkxie) [inline components/avatar/nz-avatar.component.ts]: https://github.com/NG-ZORRO/ng-zorro-antd/blob/a5e09fa4439e3c9f465e90efff72e311a9d23e23/components/button/nz-button.component.ts#L118
更新host class可以使用这个service，会更加方便

File: components/avatar/nz-avatar.component.ts [hunk 2]
```diff
@@ -1,39 +1,29 @@
...
+import { NzUpdateHostClassService } from '../core/services/update-host-class.service';
 
 export type NzAvatarShape = 'square' | 'circle';
 export type NzAvatarSize = 'small' | 'large' | 'default';
 
 @Component({
-  selector           : 'nz-avatar',
+  selector: 'nz-avatar',
+  template: `
+  <i class="anticon anticon-{{nzIcon}}" *ngIf="nzIcon && hasIcon"></i>
```
Comment (vthinkxie) [inline components/avatar/nz-avatar.component.ts]: https://github.com/NG-ZORRO/ng-zorro-antd/blob/a5e09fa4439e3c9f465e90efff72e311a9d23e23/components/steps/nz-step.component.ts#L23

nzIcon 应该直接指定 i 的className，用户可以使用自己引入的iconfont

Comment (cipchk) [reply components/avatar/nz-avatar.component.ts]: 这会倒置无法向后兼容

Comment (vthinkxie) [reply components/avatar/nz-avatar.component.ts]: 已经有无数个break changes了，不用向后兼容了

**Pull request merged**